### ISSUE TITLE
Code split more of the frontend to reduce initial app size

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#4665]
We have the capability to lazily load parts of the frontend, which we do for a couple things (`ga-metadata` and `_internal` routes) but could do a **lot** more. I don't think we need to go too crazy but there are a few things that might make a big difference:

* Admin panel - most users won't need this often or at all. It's probably huge compared to the rest of the app / dependencies
  * Ideally including reducers: https://github.com/insin/react-examples/tree/master/code-splitting-redux-reducers
* Visualizations
  * Some visualizations like pin maps probably aren't super common and have large dependencies like Leaflet. Maybe we can do it automatically for all visualizations, or manually for the large ones.
  * Or maybe the entire visualization "sub-system"? if we can load it in parallel with the first query it might not even add any latency. We'd probably need to do a little work to split split it out from the rest of the app.

We can do route-based code splitting https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/guides/code-splitting.md or component-based code splitting: https://medium.com/@thejameskyle/react-loadable-2674c59de178
